internal/services/mssql/mssql_database_resource.go

```diff
@@ -769,127 +802,135 @@ func resourceMsSqlDatabaseUpdate(d *pluginsdk.ResourceData, meta interface{}) er
 		}
 	}
 
 	if d.HasChanges("read_scale") {
-		readScale := sql.DatabaseReadScaleDisabled
+		readScale := databases.DatabaseReadScaleDisabled
 		if v := d.Get("read_scale").(bool); v {
-			readScale = sql.DatabaseReadScaleEnabled
+			readScale = databases.DatabaseReadScaleEnabled
 		}
-		props.ReadScale = readScale
+		props.ReadScale = pointer.To(readScale)
 	}
 
 	if d.HasChange("restore_point_in_time") {
 		if v, ok := d.GetOk("restore_point_in_time"); ok {
-			if createMode != string(sql.CreateModePointInTimeRestore) {
-				return fmt.Errorf("'restore_point_in_time' is supported only for create_mode %s", string(sql.CreateModePointInTimeRestore))
+			if createMode != string(databases.CreateModePointInTimeRestore) {
+				return fmt.Errorf("'restore_point_in_time' is supported only for create_mode %s", string(databases.CreateModePointInTimeRestore))
 			}
-			restorePointInTime, err := time.Parse(time.RFC3339, v.(string))
+			_, err := time.Parse(time.RFC3339, v.(string))
 			if err != nil {
 				return fmt.Errorf("parsing `restore_point_in_time` value %q for %s: %+v", v, id, err)
 			}
-			props.RestorePointInTime = &date.Time{Time: restorePointInTime}
+			props.RestorePointInTime = pointer.To(v.(string))
 		}
 	}
 
 	if d.HasChange("sku_name") {
-		payload.Sku = &sql.Sku{
-			Name: pointer.To(skuName),
-		}
+		payload.Sku = pointer.To(databases.Sku{
+			Name: skuName,
+		})
 	}
 
 	if d.HasChange("recover_database_id") {
-		props.RecoverableDatabaseID = pointer.To(d.Get("recover_database_id").(string))
+		props.RecoverableDatabaseId = pointer.To(d.Get("recover_database_id").(string))
 	}
 
 	if d.HasChange("restore_dropped_database_id") {
-		props.RestorableDroppedDatabaseID = pointer.To(d.Get("restore_dropped_database_id").(string))
+		props.RestorableDroppedDatabaseId = pointer.To(d.Get("restore_dropped_database_id").(string))
 	}
 
-	payload.DatabaseProperties = pointer.To(props)
+	payload.Properties = pointer.To(props)
 
-	future, err := client.Update(ctx, id.ResourceGroup, id.ServerName, id.Name, payload)
+	err = client.UpdateThenPoll(ctx, id, payload)
 	if err != nil {
 		return fmt.Errorf("updating %s: %+v", id, err)
 	}
 
-	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
-		return fmt.Errorf("waiting for update of %s: %+v", id, err)
-	}
-
 	// Wait for the ProvisioningState to become "Succeeded"
 	log.Printf("[DEBUG] Waiting for %s to become ready", id)
 	pendingStatuses := make([]string, 0)
-	for _, s := range sql.PossibleDatabaseStatusValues() {
-		if s != sql.DatabaseStatusOnline {
-			pendingStatuses = append(pendingStatuses, string(s))
+	for _, s := range databases.PossibleValuesForDatabaseStatus() {
+		if s != string(databases.DatabaseStatusOnline) {
+			pendingStatuses = append(pendingStatuses, s)
 		}
 	}
+
+	deadline, ok := ctx.Deadline()
+	if !ok {
+		return fmt.Errorf("internal-error: context had no deadline")
+	}
+
 	stateConf := &pluginsdk.StateChangeConf{
```

Comment:
internal x-ref: another instance of https://github.com/hashicorp/go-azure-sdk/issues/307

Reply:
Note added, fixed.